genkit-tools/common/src/manager/manager.ts

```diff
@@ -735,15 +743,24 @@ export class RuntimeManager {
   /**
    * Handles an HTTP error.
    */
-  private httpErrorHandler(error: AxiosError, message?: string): any {
+  private httpErrorHandler(error: AxiosError, message?: string): never {
     const newError = new GenkitToolsError(message || 'Internal Error');
 
     if (error.response) {
-      if ((error.response?.data as any).message) {
-        newError.message = (error.response?.data as any).message;
-      }
       // we got a non-200 response; copy the payload and rethrow
       newError.data = error.response.data as GenkitError;
+      newError.stack = (error.response?.data as any).message;
+      console.log('error.response?.data ', error.response?.data);
+      if ((error.response?.data as any).message) {
+        newError.data.data = {
+          ...newError.data.data,
+          genkitErrorMessage: message,
```

Comment:
I think it should be:

`genkitErrorMessage: (error.response?.data as any).message` (same as genkitErrorDetails.stack)

Then the UI can pick it up automatically:
<img width="1224" height="189" alt="Screenshot 2025-12-17 at 4 58 58 PM" src="https://github.com/user-attachments/assets/5859e2bf-5b7c-47a9-af06-9aa8e1ed2e49" />

this is in `src/app/utils/error.ts`, isSchemaValidationError() method in genkit UI.